### PR TITLE
Write end-to-end tests for navbar, cache yarn, node_modules, and Playwright dependencies to speed up CI

### PR DESCRIPTION
**Pull request by sandylcruz**
# Changes made

This PR adds end-to-end testing with Playwright for the navigation bar. Additionally, caching of dependencies has been added to both the linting and testing workflows.

# Setting up Playwright

## Running the dev server vs testing against Vercel preview

The testing workflow for `/docs` is different than it is in [`/next`](https://github.com/gravitational/next/pull/1374) and [`/blog`](https://github.com/gravitational/blog/pull/241). In the other repositories, the testing happens simultaneously as linting/type-checking by spinning up a [development server](https://github.com/gravitational/next/blob/f0b715f75f31bafb1cadc8273ded265e03b87ee2/playwright.config.ts#L15) to run the tests against. 

With this PR, testing happens sequentially [after Vercel build has completed](https://vercel.com/guides/how-can-i-run-end-to-end-tests-after-my-vercel-preview-deployment). This was done because there is an issue with running the dev server for testing because of this repository's use of submodules. In the future, this could be modified to speed up CI.

Because the Playright workflow is dependent on `deployment_status` of Vercel, the dependencies in the `playright` workflow cannot be cached in Github Actions. Github actions gives an error that `deployment_status` is not valid because it's not tied to a branch or ref due to this.

## Parallelism and splitting tests into shards with matrix

In order to decrease testing time, tests within the same file are being run [in parallel](https://playwright.dev/docs/test-parallel#parallelize-tests-in-a-single-file) for the navigation test file this PR introduces. Additionally, tests are split into 3 different [shards](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) to split the load between 3 workers. Three was chosen arbitrarily. The main intent is that while we write more tests, we have an easy way to scale.

## Single vs multiple browser tests

Currently the browsers used for end-to-end testing are desktop Chrome / Safari and mobile Chrome / Safari. Some tests should be run in all 4 browsers, such as checking the navigation bar functionality. In some cases, we might just want to run tests in a single browser. In other cases, the test might not even execute in the browser and would be an [API test](https://playwright.dev/docs/test-api-testing). The pattern established in this PR is that there are 3 folders: 

- All-browsers (run in all browsers)
- Single browser (just run in Chrome)
- API (does not run in any browser, but technically the configuration indicates it's running in Chrome)

# Tests added

- `/all-browsers/navigation.spec.ts` tests that functionality of navigation behaves properly and expected `<a>` are present
- `/api-tests/navigation.spec.ts` makes a `HEAD` request to all navigation routes and fails if any do not return a `200` response

## Before CI caching (of `base.yml`)

Screen Shot 2022-08-24 at 3 22 46 PM

## After CI caching

Screen Shot 2022-08-24 at 3 26 20 PM

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v14.18.1
```
Comment (jeffgaynor) [inline .nvmrc:1]: Add missing newline

File: components/Header/HeaderCTA.tsx [hunk 2]
```diff
@@ -68,8 +70,9 @@ const HeaderCTA = () => {
         </div>
         <Button
           as="link"
-          href="/pricing/"
+          href="https://goteleport.com/pricing/"
```
Comment (jeffgaynor) [inline components/Header/HeaderCTA.tsx:73]: Why adding host? Generally absolute paths are best, so they are domain agnostic.

File: components/Menu/structure.ts [hunk 3]
```diff
@@ -18,218 +20,231 @@ const menu: MenuCategoryProps[] = [
   {
     title: "Products",
     description: "Teleport Products",
-    href: "/teleport/",
+    href: `https://${HOST}/`,
```
Comment (jeffgaynor) [inline components/Menu/structure.ts:23]: What is the reasoning for adding the full host in this file?

**Review by jeffgaynor — CHANGES_REQUESTED** [review]

File: .nvmrc (hunk 1 above)
Comment (sandylcruz) [reply .nvmrc:1]: Updating

File: components/Menu/structure.ts (hunk 3 above)
Comment (sandylcruz) [reply components/Menu/structure.ts:23]: Removed

File: components/Header/HeaderCTA.tsx (hunk 2 above)
Comment (sandylcruz) [reply components/Header/HeaderCTA.tsx]: I've updated this href to be `https://${HOST}/pricing/` to align with [`structure`](https://github.com/gravitational/docs/blob/8902141643a26f692b8f79507e2155c5e6c8a636/components/Menu/structure.ts#L31) of the other external links in navigation bar.

Comment (jeffgaynor) [reply components/Header/HeaderCTA.tsx]: @sandylcruz What is the reason that the structure there though? This means that localhost and goteleport.dev links are incorrect, as an example. If there isn't

Comment (sandylcruz) [reply components/Header/HeaderCTA.tsx]: Updated

File: package.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
     "build": "yarn preflight && next build",
     "update-and-build": "yarn git-update && yarn build && yarn generate-sitemap",
     "start": "next start",
-    "test": "loadr -- uvu",
+    "test": "npx playwright test",
```
Comment (jeffgaynor) [inline package.json:17]: `playwright test` since it's now a dev dependency.

**Review by jeffgaynor — CHANGES_REQUESTED** [review]
Looks good, just one small thing.

File: package.json (hunk 4 above)
Comment (sandylcruz) [reply package.json]: Updated

**Review by jeffgaynor — APPROVED** [review]

**Pull request merged**